### PR TITLE
Update to Swift 4 (#5)

### PR DESCRIPTION
**Pull request by saagarjha**

### CODE REVIEW

File: Parser/Preferences.swift [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+        }
+    }
+    var areSemicolonsRemoved = false
+}
+
+#if swift(>=4)
+    // We can't do this yet, so fallback on the older workaround
+    // extension Preferences: Codable { }
+#else
+    protocol Codable { }
```
Comment (Jintin) [inline Parser/Preferences.swift]: great solution!

**Comment by Jintin** [comment]
please rebase again. XD

**Comment by saagarjha** [comment]
I figured out why the tests are broken: you're misusing indices [here](https://github.com/Jintin/Swimat/blob/extension/Parser/SwiftParser.swift#L294). Swift does not allow the use of a `String.Index` on any `String` but the one it originated from; here, however, you're storing it and using it on another `String` later. This is "undefined behavior" and is not guaranteed to work, as we're seeing the tests. Please fix this so it only uses legal operations; once you've finished I'll rebase it into my current branch so it can be merged.

**Comment by Jintin** [comment]
I update but still fail in xcode8 
Could you take a look of it.
https://travis-ci.org/Jintin/Swimat/builds/275713544

**Comment by saagarjha** [comment]
@Jintin There's a bug in Xcode 8's Swift compiler that was fixed in 8.3. I'd suggest removing it and keeping Xcode 8.3 and Xcode 9 only.

**Comment by Jintin** [comment]
Updated.

**Comment by saagarjha** [comment]
Great, let me rebase my branch then.

**Comment by saagarjha** [comment]
@Jintin I've rebased this branch.

File: .travis.yml [hunk 2]
```diff
@@ -1,7 +1,6 @@
 language: objective-c
 matrix:
   include:
-    - osx_image: xcode8.3
```
Comment (Jintin) [inline .travis.yml:4]: Can't we test on xcode8.3?

Comment (saagarjha) [reply .travis.yml:4]: No, since Xcode 8.3 doesn’t support Swift 4.

**Pull request merged**